src/components/QuestionRow.tsx

```diff
@@ -0,0 +1,198 @@
+        if (z.string().url().safeParse(text).success) {
+          pasteText(textAreaRefs.current[1], `[link](${text})`);
+        } else {
+          pasteText(textAreaRefs.current[1], text);
+        }
+        onQuestionChange({ ...question, body: textAreaRefs.current[1]!.value });
+        return;
+      } else if (types === 'Files') {
+        const file = e.clipboardData.files[0];
+        if (file?.type.startsWith('image/')) {
+          const res = (await getPresignedUrl.refetch()).data;
+          if (res) {
+            const { url, fields } = res;
+            const formData = new FormData();
+            Object.entries({
+              ...fields,
+              "Content-Type": file.type,
+              file,
+            }).forEach(([key, value]) => {
+              formData.append(key, value);
+            });
+            void fetch(url, {
+              method: "POST",
+              body: formData,
+            });
+            pasteText(textAreaRefs.current[1], `![image](${url}/${fields.key})`);
+            onQuestionChange({ ...question, body: textAreaRefs.current[1]!.value });
+          }
+          return;
+        }
+      }
+    }
+  }
+
+  // const handleKey = (e: KeyboardEvent<HTMLTextAreaElement>) => {
+  //   if (e.ctrlKey) {
+  //     switch (e.key) {
+  //       case 'B':
+  //         e.stopPropagation();
+  //         bracket(textAreaRefs.current[1], '**');
+  //         break;
+  //       case 'I':
+  //         e.stopPropagation();
+  //         bracket(textAreaRefs.current[1], '*');
+  //         break;
+  //     }
+  //   }
+  // };
+
+  useEffect(() => {
+    void (async () => setHTML((await parseMD(body))))();
+    // eslint-disable-next-line
+  }, []);
+
+  useEffect(() => {
+    if (debouncedBodyState === EDITING && textAreaRefs.current[1] !== document.activeElement) {
+      textAreaRefs.current[1]?.focus();
+    } else if (debouncedBodyState !== EDITING && prevBodyState.current === EDITING) {
+      void (async () => setHTML((await parseMD(body))))();
+    }
+    prevBodyState.current = debouncedBodyState;
+    // eslint-disable-next-line
+  }, [debouncedBodyState]);
+
+  useLayoutEffect(() => {
+    textAreaRefs.current.forEach(r => {
+      if (r) {
+        r.style.height = `${MIN_TEXTAREA_HEIGHT_px}px`;
+      }
+    });
+    let maxHeight = Infinity;
+    if (bodyState === COLLAPSED) {
+      maxHeight = MAX_TEXTAREA_HEIGHT_px;
+    } else if (bodyState === EDITING) {
+      maxHeight = MAX_TEXTAREA_EDITING_HEIGHT_px;
+    }
+    const height = Math.min(Math.max(
+      ...textAreaRefs.current.map(r => r?.scrollHeight ?? 0),
+      bodyState === EDITING ? 0 : rendererRef.current!.scrollHeight + 1,
+      MIN_TEXTAREA_HEIGHT_px
+    ), maxHeight);
+    textAreaRefs.current.forEach(r => {
+      if (r) {
+        r.style.height = `${height}px`;
+      }
+    });
+    rendererRef.current!.style.height = `${height}px`;
+  }, [body, html, bodyState]);
+
+  return <div {...others}>
```

Comment:
![errornousBody](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g39/assets/70463178/144943e6-08a2-43ec-b2ab-3ec0da07449d)

Experiencing this glitch trying to edit the body area

Reply:
This bug only appears when I'm not using data.zip though

Reply:
seems to be a chrome specific bug, firefox works fine

Reply:
I can't seem to replicate it. Can you try double clicking instead?

Reply:
I think its fixed, not too sure what fixed it though. I managed to replicate the bug on 5aadffdad98e8adc5494d2e0ab64c0897567fb31
I am double clicking in the gif.

Reply:
I've added it as a resolved issue for doc, in case it happens again.